tests/openebs/storage_status.go

```diff
@@ -170,3 +171,71 @@ func getPoolNameFromCVR(k v1alpha1.CStorVolumeReplica) string {
 func getVolumeNameFromCVR(k v1alpha1.CStorVolumeReplica) string {
 	return k.Labels[cVRPVLabel]
 }
+
+// GetCStorBackups returns cstorbackup list for the given backup
+func (c *ClientSet) GetCStorBackups(backup, ns string) (*v1alpha1.CStorBackupList, error) {
+	return c.OpenebsV1alpha1().
+		CStorBackups(ns).
+		List(metav1.ListOptions{
+			LabelSelector: "openebs.io/backup=" + backup,
+		})
+}
+
+// GetCStorCompletedBackups returns cstorcompletedbackup list for the given backup
+func (c *ClientSet) GetCStorCompletedBackups(backup, ns string) (*v1alpha1.CStorCompletedBackupList, error) {
+	return c.OpenebsV1alpha1().
+		CStorCompletedBackups(ns).
+		List(metav1.ListOptions{
+			LabelSelector: "openebs.io/backup=" + backup,
+		})
+}
+
+// IsBackupResourcesExist checks if backupResources, for the given backup, exist or not
+func (c *ClientSet) IsBackupResourcesExist(backup, pvc, ns string) (bool, error) {
+	isSchedule := false
+	isLastBackup := false
+
+	scheduleName := backup
+	splitName := strings.Split(backup, "-")
+	if len(splitName) >= 2 {
+		isSchedule = true
+		scheduleName = strings.Join(splitName[0:len(splitName)-1], "-")
+	}
+
+	blist, err := c.GetCStorBackups(scheduleName, ns)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to fetch cstorbackup list for backup %s/%s", ns, backup)
+	}
+
+	cblist, err := c.GetCStorCompletedBackups(scheduleName, ns)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to fetch cstorcompletedbackup list for backup %s/%s", ns, backup)
+	}
+
+	if isSchedule && len(cblist.Items) == 1 {
```

Comment:
This section could use a comment on why the last backup should not be deleted. I believe it is because of the ability to send incremental snapshots. But I am not sure. And in case it is for incremental snapshots, does this code get executed only for scheduled snapshots?

Reply:
yes. it will be executed in case scheduled snapshots only. I'll add comments and more checks